### ISSUE TITLE
Normalising flows need training tracking

### ISSUE THREAD

**Issue by ataras2** [benjaminpope/sibylla#14]
Training and validation loss need to be analyzed easily each run and saved with the model itself by default

**Comment by ataras2** [comment]
Implemented and also includes the functionality to use multiple different validation sets
